server/routes/product.js

```diff
@@ -138,17 +157,59 @@ router.get("/:shop_id/products", async (req, res) => {
     try {
         let id = req.params.shop_id;
 
-        if (!mongoose.Types.ObjectId.isValid(id)){
+        if (!mongoose.Types.ObjectId.isValid(id)) {
             return res.status(400).json({ msg: "id not valid" });
         }
 
-        let products = await Product.find({ userId: id })
+        let products = await Product.find({ userId: id });
 
         res.json(products);
+    } catch (err) {
+        res.status(500).send("Server Error");
+    }
+});
+
+// @route     POST product/pagination/products
+// @desc      Get specific page products based on client search
+// @access    Public
+router.post("/pagination/products", async (req, res) => {
+    let limit = req.body.limit ? parseInt(req.body.limit) : 20;
```

Comment:
save `req.body` in a variable so you don't need to reach into this object multiple times in this controller